src/operation/run_command/mod.rs

```diff
@@ -109,7 +110,16 @@ impl super::Response for Response {
             .ok()
             .and_then(|doc| bson::from_document(doc.clone()).ok());
 
-        Ok(Self { doc, cluster_time })
+        let recovery_token = doc
+            .get_document("recoveryToken")
+            .ok()
+            .and_then(|doc| bson::from_document(doc.clone()).ok());
```

Comment:
The conversion on this line shouldn't be necessary as the type for `recovery_token` is already `Option<Document>`.